### PR TITLE
Started a fresh and simpler implementation

### PR DESCRIPTION
**Pull request by jentfoo**
Things which were done:
    \* gradle build system added
    \* Removed dependency on log4j
    \* Removed executor that was used internally
    \* Changed how the response object is build (now using an interface for configurable values)
        - This includes no longer trying to fill in data from an actual servlet request
    \* added the first and very basic unit test around the json builder

The things which are not unit tested were not changed, or are incredibly simple. But let me know if you feel like we should add more (bit excited to start using this).

### CODE REVIEW

**Comment by blendmaster** [comment]
NDOTM. I'd rather use jackson databind and okhttp/apachehttp for keepalive support and overall less jankiness, but what you have probably works, and I find it hard to get excited about logging code.

File: src/main/java/com/muantech/rollbar/java/HttpRequest.java [hunk 1]
```diff
@@ -47,44 +46,45 @@ public void setBody(String body) {
...
+            
             if (body != null) {
                 connection.setDoOutput(true);
                 writeBody(body, connection);
             }
-
-            if (connection.getResponseCode() != 200) return false;
-
+            
+            if (connection.getResponseCode() != 200) {
```
Comment (blendmaster) [inline src/main/java/com/muantech/rollbar/java/HttpRequest.java]: You shouldn't retry 4xx codes, I'd change this to `>= 500`.

**Comment by jentfoo** [comment]
yea, i find it hard too, haha.  I don't like his HTTP connection stuff, but I left it unchanged since it seems to work.  In general I just wanted something simple and minimal that we can use anywhere we want.

**Comment by blendmaster** [comment]
Oh, before I forget the most important thing: your trailing whitespace needs some serious trimming, yo.

**Comment by blendmaster** [comment]
The rest of your files still have trailing whitespace, you probably need to resave them.

**Comment by blendmaster** [comment]
Ell Gee Tee Emm. You can squash your commits if you want.

**Pull request merged**